examples/chip-tool/templates/tests/partials/value_equals.zapt

```diff
@@ -51,8 +51,18 @@
   {{else}}
     VerifyOrReturn(CheckValue{{#if (isString type)}}AsString{{/if}}("{{label}}", {{actual}}, 
       {{~#if (chip_tests_variables_has expected)}}{{expected}}
-      {{else if (isOctetString type)}}chip::ByteSpan(chip::Uint8::from_const_char("{{octetStringEscapedForCLiteral expected}}"), {{expected.length}})
-      {{else if (isCharString type)}}chip::CharSpan("{{expected}}", {{utf8StringLength expected}})
+      {{else if (isOctetString type)}}
+        {{#if keepAsExpected}}
```

Comment:
I think this should probably be more like this:
```
{{~#if (chip_tests_variables_has expected)}}{{expected}}
{{else if keepAsExpected}}{{expected}}
{{else if (isOctetString type)}}do-things-as-now
...
```
because otherwise a nullable integer type will fail too, right?